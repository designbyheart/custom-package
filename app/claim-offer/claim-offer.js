// @flow
import React, { PureComponent } from 'react'
import { StyleSheet, Platform, FlatList, View } from 'react-native'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import {
  claimOfferShown,
  acceptClaimOffer,
  claimOfferRejected,
  claimOfferIgnored,
} from '../store'
import {
  Container,
  CustomView,
  CustomButton,
  CustomText,
  Icon,
  ImageColorPicker,
  ConnectionTheme,
} from '../components'
import { homeRoute } from '../common/'
import { OFFSET_2X, OFFSET_1X, color } from '../common/styles'
import type { ClaimOfferPayload, Attribute } from './type-claim-offer'
import type { Store } from '../store/type-store'
import ClaimRequestModal from './claim-request-modal'

class ClaimOfferHeader extends PureComponent {
  render() {
    const { issuer, claimOffer }: ClaimOfferPayload = this.props.payload
    const logoUrl = issuer.logoUrl
      ? { uri: issuer.logoUrl }
      : require('../images/cb_evernym.png')

    return (
      <CustomView fifth style={[styles.header]}>
        <CustomView fifth center style={[styles.headerStripLogoContainer]}>
          <Icon
            absolute="TopRight"
            src={require('../images/close.png')}
            small
            testID="claim-offer-icon-close"
            onPress={this.props.onClose}
            iconStyle={[styles.headerCloseIcon]}
            style={[styles.headerCloseIconContainer]}
          />
          <ConnectionTheme logoUrl={issuer.logoUrl} style={[styles.strip]} />
          <Icon
            center
            halo
            extraLarge
            resizeMode="cover"
            src={logoUrl}
            style={[styles.issuerLogo]}
            testID="claim-offer-issuer-logo"
          />
        </CustomView>
        <CustomView fifth center style={[styles.issuerDetail]}>
          <CustomText h5 bg="fifth">
            {issuer.name} is offering you
          </CustomText>
          <CustomText h4 bold bg="fifth" style={[styles.claimOfferName]}>
            {claimOffer.name}
          </CustomText>
        </CustomView>
        <ImageColorPicker imageUrl={issuer.logoUrl} />
      </CustomView>
    )
  }
}

class Separator extends PureComponent {
  render() {
    return <CustomView style={[styles.separator]} />
  }
}

class ClaimOfferAttributeList extends PureComponent {
  keyExtractor = (_, index: number) => index

  renderItem = ({ item }: { item: Attribute }) => {
    return (
      <CustomView
        fifth
        row
        horizontalSpace
        doubleVerticalSpace
        style={[styles.attributeListItem]}
      >
        <CustomView fifth right style={[styles.attributeListLabel]}>
          <CustomText
            h7
            uppercase
            bg="fifth"
            style={[styles.attributeListLabelText]}
          >
            {item.label}
          </CustomText>
        </CustomView>
        <CustomView fifth left style={[styles.attributeListValue]}>
          <CustomText h6 bold bg="fifth">
            {item.data}
          </CustomText>
        </CustomView>
      </CustomView>
    )
  }

  render() {
    const attributes: Array<Attribute> = this.props.list
    return (
      <FlatList
        data={attributes}
        ItemSeparatorComponent={Separator}
        keyExtractor={this.keyExtractor}
        ListFooterComponent={Separator}
        renderItem={this.renderItem}
      />
    )
  }
}

export class ClaimOffer extends PureComponent {
  close = () => {
    this.props.navigation.goBack()
  }

  onIgnore = () => {
    this.props.claimOfferIgnored()
    this.close()
  }

  onReject = () => {
    this.props.claimOfferRejected()
    this.close()
  }

  onAccept = () => {
    this.props.acceptClaimOffer()
  }

  componentDidMount() {
    // update store that offer is shown to user
    this.props.claimOfferShown()
  }

  render() {
    const { payload, claimRequestStatus } = this.props.claimOffer
    const isValid =
      payload &&
      payload.claimOffer &&
      payload.issuer &&
      payload.claimOffer.revealedAttributes

    const { issuer: { logoUrl } }: ClaimOfferPayload = payload

    return (
      <Container fifth>
        {isValid && (
          <ClaimOfferHeader payload={payload} onClose={this.onIgnore} />
        )}
        {isValid ? (
          <Container fifth style={[styles.claimOfferData]}>
            <ClaimOfferAttributeList
              list={payload.claimOffer.revealedAttributes}
            />
          </Container>
        ) : (
          <Container fifth center>
            <CustomText h5 bg="fifth">
              Invalid claim offer. Please ignore.
            </CustomText>
          </Container>
        )}
        <CustomView row>
          <Container>
            <ConnectionTheme logoUrl={logoUrl} secondary>
              <CustomButton raised title="Ignore" onPress={this.onReject} />
            </ConnectionTheme>
          </Container>
          <Container>
            <ConnectionTheme logoUrl={logoUrl}>
              <CustomButton raised title="Accept" onPress={this.onAccept} />
            </ConnectionTheme>
          </Container>
        </CustomView>
        {isValid && (
          <ClaimRequestModal
            claimRequestStatus={claimRequestStatus}
            payload={payload}
            onContinue={this.close}
          />
        )}
      </Container>
    )
  }
}

const mapStateToProps = ({ claimOffer }: Store) => ({
  claimOffer,
})

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      claimOfferShown,
      acceptClaimOffer,
      claimOfferRejected,
      claimOfferIgnored,
    },
    dispatch
  )

export default connect(mapStateToProps, mapDispatchToProps)(ClaimOffer)

const styles = StyleSheet.create({
  header: {
    paddingTop: (Platform.OS === 'ios' ? OFFSET_2X : 0) + 14,
    shadowColor: color.bg.secondary.color,
    shadowOffset: { width: 0, height: 10 },
    shadowOpacity: 0.3,
    shadowRadius: 10,
    zIndex: 4,
  },
  headerCloseIcon: {
    marginRight: 15,
  },
  headerCloseIconContainer: {
    zIndex: 2,
  },
  headerStripLogoContainer: {
    height: 90,
  },
  strip: {
    height: 20,
    width: '100%',
    backgroundColor: color.actions.secondary,
    zIndex: -1,
  },
  issuerLogo: {
    ...StyleSheet.absoluteFillObject,
    alignSelf: 'center',
    height: 80,
    zIndex: 1,
  },
  issuerDetail: {
    marginTop: OFFSET_1X / 2,
    paddingBottom: OFFSET_1X,
  },
  claimOfferName: {
    marginTop: OFFSET_1X / 2,
  },
  claimOfferData: {
    zIndex: 3,
  },
  attributeListLabel: {
    flex: 1,
    paddingRight: OFFSET_1X,
  },
  attributeListLabelText: {
    lineHeight: 14,
  },
  attributeListValue: {
    flex: 2,
  },
  separator: {
    height: 1,
    backgroundColor: color.bg.secondary.font.tertiary,
  },
})
