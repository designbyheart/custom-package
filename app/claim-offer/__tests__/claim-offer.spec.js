// @flow
import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import { ClaimOffer } from '../claim-offer'

describe('<ClaimOffer />', () => {
  it('should call claimOfferShown', () => {
    const claimOfferShown = jest.fn()
    const wrapper = renderer
      .create(<ClaimOffer claimOfferShown={claimOfferShown} />)
      .toJSON()
    expect(wrapper).toMatchSnapshot()
    expect(claimOfferShown).toHaveBeenCalled()
  })
})
