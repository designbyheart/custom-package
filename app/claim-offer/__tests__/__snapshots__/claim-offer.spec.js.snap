// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClaimOffer /> should call claimOfferShown on componentDidMount 1`] = `
<View
  fifth={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View
    fifth={true}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "paddingTop": 34,
        "shadowColor": "#777777",
        "shadowOffset": Object {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 10,
        "zIndex": 4,
      }
    }
  >
    <View
      center={true}
      fifth={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "height": 90,
          "justifyContent": "center",
        }
      }
    >
      <View
        absolute="TopRight"
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        clearBg={true}
        hitSlop={undefined}
        iconStyle={
          Array [
            Object {
              "marginRight": 15,
            },
          ]
        }
        onLayout={undefined}
        onPress={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        small={true}
        src={1}
        style={
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 2,
          }
        }
        testID="claim-offer-icon-close-touchable"
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {
                "height": 20,
                "width": 20,
              },
              Object {
                "marginRight": 15,
              },
            ]
          }
          testID="claim-offer-icon-close"
        />
      </View>
      <View
        style={
          Array [
            Array [
              Object {
                "backgroundColor": "#f79347",
                "height": 20,
                "width": "100%",
                "zIndex": -1,
              },
            ],
            Object {
              "backgroundColor": "rgba(133, 191, 67, 1.0)",
            },
          ]
        }
      />
      <View
        center={true}
        clearBg={true}
        extraLarge={true}
        halo={true}
        resizeMode="cover"
        src={
          Object {
            "uri": "http://testissuer.com/logoUrl.png",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "bottom": 0,
            "height": 80,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
        testID="claim-offer-issuer-logo"
      >
        <View
          center={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.07)",
              "borderRadius": 50,
              "height": 100,
              "justifyContent": "center",
              "width": 100,
            }
          }
        >
          <Image
            resizeMode="cover"
            source={
              Object {
                "uri": "http://testissuer.com/logoUrl.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 80,
                  "width": 80,
                },
              ]
            }
            testID="claim-offer-issuer-logo"
          />
        </View>
      </View>
    </View>
    <View
      center={true}
      fifth={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "justifyContent": "center",
          "marginTop": 5,
          "paddingBottom": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={undefined}
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "SF Pro Display",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            null,
            null,
            null,
          ]
        }
        testID={undefined}
      >
        Test Issuer
         is offering you
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={undefined}
        style={
          Array [
            Object {
              "fontSize": 20,
            },
            Object {
              "fontFamily": "SF Pro Display",
            },
            Object {
              "backgroundColor": "#ffffff",
              "color": "#777777",
            },
            Object {
              "fontWeight": "bold",
            },
            null,
            null,
            Object {
              "marginTop": 5,
            },
          ]
        }
        testID={undefined}
      >
        Home Address
      </Text>
    </View>
    <WebView
      javaScriptEnabled={true}
      onMessage={[Function]}
      source={
        Object {
          "html": "
                <!DOCTYPE html>
                <html>
                <script>
                function fetchColorFromCanvas(){
                  var rgb = getAverageRGB(document.getElementById(\\"canvasImage\\"));
                  var message = {\\"message\\":'rgbValues',\\"payload\\":rgb};
                  window.postMessage(JSON.stringify(message));
                }
        
                function getAverageRGB(imgEl) {
                  var blockSize = 5;
                  var defaultRGB = {r:0,g:0,b:0};
                  var canvas = document.createElement(\\"canvas\\");
                  var context = canvas.getContext && canvas.getContext(\\"2d\\");
                  var data, width, height;
                  var i = -4;
                  var length;
                  var rgb = {r:0,g:0,b:0};
                  var count = 0;
                  if (context == undefined || context == null) {
                      return defaultRGB;
                  }
                  height = canvas.height = imgEl.naturalHeight || imgEl.offsetHeight || imgEl.height;
                  width = canvas.width = imgEl.naturalWidth || imgEl.offsetWidth || imgEl.width;
                  context.drawImage(imgEl, 0, 0);
                  try {
                      data = context.getImageData(width/3, width/3 + 10, width/3, width/3 + 10);
                  } catch(e) {
                      return defaultRGB;
                  }
        
                  length = data.data.length;  
                  while ( (i += blockSize * 4) < length ) {
                      ++count;
                      rgb.r += data.data[i];
                      rgb.g += data.data[i+1];
                      rgb.b += data.data[i+2];              
                  }
                  rgb.r = ~~(rgb.r/count);
                  rgb.g = ~~(rgb.g/count);
                  rgb.b = ~~(rgb.b/count);
                  return rgb;
                }
        
                const interval = setInterval(function() {
                  var img = document.getElementById('canvasImage');
                  if(img.src.length != 0) {
                    fetchColorFromCanvas();
                    clearInterval(interval);
                  }
                },10);
        
                </script>
                <body style='position: absolute; bottom: -1000px;'>
                  <img src='data:image/jpeg;base64,' id='canvasImage' onload='fetchColorFromCanvas()'/>
                </body>
                </html>",
        }
      }
      style={
        Object {
          "position": "absolute",
          "right": -1000,
        }
      }
    />
  </View>
  <View
    fifth={true}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
        "zIndex": 3,
      }
    }
  >
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      ListFooterComponent={[Function]}
      data={
        Array [
          Object {
            "data": "An Address",
            "label": "Address 1",
          },
          Object {
            "data": "An Address 2",
            "label": "Address 2",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onViewableItemsChanged={undefined}
      removeClippedSubviews={true}
      renderItem={[Function]}
      renderScrollComponent={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
        >
          <View
            doubleVerticalSpace={true}
            fifth={true}
            horizontalSpace={true}
            row={true}
            style={
              Object {
                "backgroundColor": "#ffffff",
                "flexDirection": "row",
                "paddingHorizontal": 15,
                "paddingVertical": 16,
              }
            }
          >
            <View
              fifth={true}
              right={true}
              style={
                Object {
                  "alignItems": "flex-end",
                  "backgroundColor": "#ffffff",
                  "flex": 1,
                  "paddingRight": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                onPress={undefined}
                style={
                  Array [
                    Object {
                      "fontSize": 12,
                    },
                    Object {
                      "fontFamily": "SF Pro Display",
                    },
                    Object {
                      "backgroundColor": "#ffffff",
                      "color": "#777777",
                    },
                    null,
                    null,
                    null,
                    Object {
                      "lineHeight": 14,
                    },
                  ]
                }
                testID={undefined}
              >
                ADDRESS 1
              </Text>
            </View>
            <View
              fifth={true}
              left={true}
              style={
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "#ffffff",
                  "flex": 2,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                onPress={undefined}
                style={
                  Array [
                    Object {
                      "fontSize": 15,
                    },
                    Object {
                      "fontFamily": "SF Pro Display",
                    },
                    Object {
                      "backgroundColor": "#ffffff",
                      "color": "#777777",
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                    null,
                    null,
                  ]
                }
                testID={undefined}
              >
                An Address
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#f0f0f0",
              "height": 1,
            }
          }
        />
        <View
          onLayout={[Function]}
        >
          <View
            doubleVerticalSpace={true}
            fifth={true}
            horizontalSpace={true}
            row={true}
            style={
              Object {
                "backgroundColor": "#ffffff",
                "flexDirection": "row",
                "paddingHorizontal": 15,
                "paddingVertical": 16,
              }
            }
          >
            <View
              fifth={true}
              right={true}
              style={
                Object {
                  "alignItems": "flex-end",
                  "backgroundColor": "#ffffff",
                  "flex": 1,
                  "paddingRight": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                onPress={undefined}
                style={
                  Array [
                    Object {
                      "fontSize": 12,
                    },
                    Object {
                      "fontFamily": "SF Pro Display",
                    },
                    Object {
                      "backgroundColor": "#ffffff",
                      "color": "#777777",
                    },
                    null,
                    null,
                    null,
                    Object {
                      "lineHeight": 14,
                    },
                  ]
                }
                testID={undefined}
              >
                ADDRESS 2
              </Text>
            </View>
            <View
              fifth={true}
              left={true}
              style={
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "#ffffff",
                  "flex": 2,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                onPress={undefined}
                style={
                  Array [
                    Object {
                      "fontSize": 15,
                    },
                    Object {
                      "fontFamily": "SF Pro Display",
                    },
                    Object {
                      "backgroundColor": "#ffffff",
                      "color": "#777777",
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                    null,
                    null,
                  ]
                }
                testID={undefined}
              >
                An Address 2
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
        >
          <View
            style={
              Object {
                "backgroundColor": "#f0f0f0",
                "height": 1,
              }
            }
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    row={true}
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginLeft": 15,
                  "marginRight": 15,
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "rgba(133, 191, 67, 0.5)",
                },
                undefined,
                Object {
                  "shadowColor": "rgba(0,0,0, .4)",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 1,
                },
                Object {
                  "padding": 12,
                },
                Array [
                  Object {
                    "marginLeft": 0,
                    "marginRight": 0,
                  },
                  null,
                  null,
                ],
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 16,
                    },
                    undefined,
                    Object {
                      "fontSize": 14,
                    },
                    undefined,
                    Object {
                      "fontSize": 17,
                      "fontWeight": "600",
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Ignore
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginLeft": 15,
                  "marginRight": 15,
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "rgba(133, 191, 67, 1.0)",
                },
                undefined,
                Object {
                  "shadowColor": "rgba(0,0,0, .4)",
                  "shadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 1,
                },
                Object {
                  "padding": 12,
                },
                Array [
                  Object {
                    "marginLeft": 0,
                    "marginRight": 0,
                  },
                  null,
                  null,
                ],
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 16,
                    },
                    undefined,
                    Object {
                      "fontSize": 14,
                    },
                    undefined,
                    Object {
                      "fontSize": 17,
                      "fontWeight": "600",
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Accept
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <Modal
    backdropColor="#f0f0f0"
    backdropOpacity={0.7}
    isVisible={false}
  >
    <View
      fifth={true}
      shadow={true}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "marginHorizontal": 10,
          "shadowColor": "rgba(0, 0, 0, 0.2)",
          "shadowOffset": Object {
            "height": 10,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 30,
        }
      }
    >
      <View
        hCenter={true}
        style={
          Object {
            "borderBottomColor": "#3f3f3f",
            "borderBottomWidth": NaN,
            "justifyContent": "center",
            "paddingVertical": 10,
          }
        }
      >
        <View
          row={true}
          spaceBetween={true}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginHorizontal": 20,
              "marginVertical": 10,
            }
          }
          vCenter={true}
        >
          <View
            imageShadow="imageShadow"
            style={
              Object {
                "borderRadius": 38,
                "shadowColor": "rgba(0, 0, 0, 0.25)",
                "shadowOffset": Object {
                  "height": 3,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 8,
              }
            }
            testID="claim-request-avatars-invitee"
          >
            <Image
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onError={undefined}
              onLayout={undefined}
              onLoad={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              source={1}
              style={
                Array [
                  Object {
                    "height": 76,
                    "width": 76,
                  },
                  Object {
                    "borderRadius": 38,
                  },
                ]
              }
              testID={undefined}
            />
          </View>
          <View
            imageShadow="imageShadow"
            style={
              Object {
                "borderRadius": 38,
                "shadowColor": "rgba(0, 0, 0, 0.25)",
                "shadowOffset": Object {
                  "height": 3,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 8,
              }
            }
            testID="claim-request-avatars-inviter"
          >
            <Image
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onError={undefined}
              onLayout={undefined}
              onLoad={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              source={
                Object {
                  "uri": "http://testissuer.com/logoUrl.png",
                }
              }
              style={
                Array [
                  Object {
                    "height": 76,
                    "width": 76,
                  },
                  Object {
                    "borderRadius": 38,
                  },
                ]
              }
              testID={undefined}
            />
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          onPress={undefined}
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "fontFamily": "SF Pro Display",
              },
              Object {
                "backgroundColor": "#ffffff",
                "color": "#777777",
              },
              Object {
                "fontWeight": "600",
              },
              Object {
                "textAlign": "center",
              },
              null,
              Object {
                "color": "#686868",
                "margin": 10,
              },
            ]
          }
          testID="claim-request-issuer-waiting"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          onPress={undefined}
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "fontFamily": "SF Pro Display",
              },
              Object {
                "backgroundColor": "#ffffff",
                "color": "#777777",
              },
              Object {
                "fontWeight": "bold",
              },
              Object {
                "textAlign": "center",
              },
              null,
            ]
          }
          testID="claim-request-claim-name"
        >
          Home Address
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginLeft": 15,
                "marginRight": 15,
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#ffffff",
              },
              undefined,
              undefined,
              Object {
                "padding": 12,
              },
              Array [
                Object {
                  "marginLeft": 0,
                  "marginRight": 0,
                },
                null,
                Object {
                  "backgroundColor": "#ffffff",
                  "opacity": 0.2,
                },
              ],
              Object {
                "backgroundColor": "#dadee0",
              },
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#85bf43",
                  },
                  Object {
                    "fontSize": 14,
                  },
                  undefined,
                  Object {
                    "fontSize": 17,
                    "fontWeight": "600",
                  },
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Continue
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
